### PR TITLE
[rospy_tutorials] Add example of periodical publishing with rospy.Timer

### PR DESCRIPTION
**Pull request by wkentaro**
As suggested in https://github.com/ros/ros_comm/pull/864#issuecomment-241907216

### CODE REVIEW

File: rospy_tutorials/001_talker_listener/talker_timer.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    hello_str = "hello world %s" % event.current_real.to_sec()
+    rospy.loginfo(hello_str)
+    pub.publish(hello_str)
+
+
+if __name__ == '__main__':
+    try:
+        rospy.init_node('talker', anonymous=True)
+        pub = rospy.Publisher('chatter', String, queue_size=10)
+        timer = rospy.Timer(rospy.Duration(1. / 10), publish_callback)  # 10Hz
```
Comment (dirk-thomas) [inline rospy_tutorials/001_talker_listener/talker_timer.py:51]: A `Timer` expects a `Time` as the first argument. While passing a `Duration` works it is less intuitive imo.

**Comment by wkentaro** [comment]
Fixed.

File: rospy_tutorials/001_talker_listener/talker_timer.py (hunk 1 above)
Comment (dirk-thomas) [reply rospy_tutorials/001_talker_listener/talker_timer.py:51]: My comment was obviously wrong. Passing `Time` doesn't work and the test fails with a stacktrace. But somehow the PR job didn't detect the problem.

I will look into the job. Once it reports the failure correctly we can make it pass by removing the second commit (which I asked for).

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

File: rospy_tutorials/test/test-talker-listener-with-timer.launch [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<launch>
+  <node name="talker" pkg="rospy_tutorials" type="talker_timer.py" />
+  <test test-name="talker_listener_test" pkg="rospy_tutorials" type="talker_listener_test.py" />
```
Comment (dirk-thomas) [inline rospy_tutorials/test/test-talker-listener-with-timer.launch]: The `test-name` is the same as for another test. Please add a suffix (`_with_timer`) to it in order to make it unique.

**Comment by wkentaro** [comment]
Fixed.

File: rospy_tutorials/test/test-talker-listener-with-timer.launch [hunk 3]
```diff
@@ -1,4 +1,9 @@
 <launch>
+
   <node name="talker" pkg="rospy_tutorials" type="talker_timer.py" />
-  <test test-name="talker_listener_test" pkg="rospy_tutorials" type="talker_listener_test.py" />
+
+  <test test-name="talker_listener_test_with_timer"
+        name="talker_listener_test_with_timer"
```
Comment (dirk-thomas) [inline rospy_tutorials/test/test-talker-listener-with-timer.launch]: This shouldn't have a `name` attribute.

Comment (wkentaro) [reply rospy_tutorials/test/test-talker-listener-with-timer.launch]: So, should the `name` be specified when the node name is used in testing to resolve name of private topic name or param name?

Comment (dirk-thomas) [reply rospy_tutorials/test/test-talker-listener-with-timer.launch]: I don't know - you might want to try it.

But this test doesn't have anything like that.

**Comment by dirk-thomas** [comment]
Great, now the PR job detected the failing test correctly. Can you please revert the second commit now and use `Duration` instead of `Time` again.

**Comment by wkentaro** [comment]
Fixed.

File: rospy_tutorials/001_talker_listener/talker_timer.py (hunk 1 above)
Comment (wkentaro) [reply rospy_tutorials/001_talker_listener/talker_timer.py:51]: Then, is this line wrong?
https://github.com/ros/ros_comm/blob/kinetic-devel/clients/rospy/src/rospy/timer.py#L188

Comment (wkentaro) [reply rospy_tutorials/001_talker_listener/talker_timer.py:51]: The wrong docstring will be fixed via https://github.com/ros/ros_comm/pull/878.

**Comment by dirk-thomas** [comment]
Thank you for iterating on this and fixing the documentation.

**Comment by wkentaro** [comment]
Thanks!

**Pull request merged**